Uplift/Views/DropdownView.swift

```diff
@@ -0,0 +1,190 @@
+            halfView.addGestureRecognizer(collapseHalfDropdownGesture)
+        }
+        
+        self.halfCollapseViewHeight = halfCollapseViewHeight
+        
+        self.halfHeight = halfHeight
+        
+        setupConstraints()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func setupConstraints() {
+        NSLayoutConstraint.activate([
+            headerView.topAnchor.constraint(equalTo: topAnchor),
+            headerView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            headerView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            headerView.heightAnchor.constraint(equalToConstant: headerViewHeight),
+            contentView.topAnchor.constraint(equalTo: headerView.bottomAnchor),
+            contentView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            contentView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        ])
+        
+        if halfDropdownEnabled {
+            NSLayoutConstraint.activate([
+                halfExpandView!.topAnchor.constraint(equalTo: contentView.bottomAnchor),
+                halfExpandView!.leadingAnchor.constraint(equalTo: leadingAnchor),
+                halfExpandView!.trailingAnchor.constraint(equalTo: trailingAnchor),
+                halfExpandView!.heightAnchor.constraint(equalToConstant: halfExpandViewHeight),
+                halfCollapseView!.topAnchor.constraint(equalTo: contentView.bottomAnchor),
+                halfCollapseView!.leadingAnchor.constraint(equalTo: leadingAnchor),
+                halfCollapseView!.trailingAnchor.constraint(equalTo: trailingAnchor),
+                halfCollapseView!.heightAnchor.constraint(equalToConstant: halfCollapseViewHeight)
+            ])
+        }
+        
+        contentViewHeightConstraint = contentView.heightAnchor.constraint(equalToConstant: 0)
+        contentViewHeightConstraint.isActive = true
+    }
+    
+    @objc func expandCollapseDropdown(sender: UITapGestureRecognizer) {
+        switch status{
+        case .up:
+            if halfDropdownEnabled {
+                status = .half
+                halfExpandView?.isHidden = false
+                UIView.animate(withDuration: 0.3) {
+                    self.contentViewHeightConstraint.constant = CGFloat(self.halfHeight)
+                    self.layoutIfNeeded()
+                }
+                currentHeight = headerViewHeight + halfExpandViewHeight + halfHeight
+                delegate.expandDropdownViewHalf(sender: self)
+            } else {
+                status = .down
+                UIView.animate(withDuration: 0.3) {
+                    self.contentViewHeightConstraint.constant = CGFloat(self.contentViewHeight)
+                    self.layoutIfNeeded()
+                }
+                currentHeight = headerViewHeight + contentViewHeight
+                delegate.expandDropdownViewFull(sender: self)
+            }
```

Comment:
can we add comments for each `if/else` block representing what being in that block means?